### ISSUE TITLE
Support for multiple service names for Mule apps running in the same JVM

### ISSUE THREAD

**Issue by michaelhyatt** [michaelhyatt/elastic-apm-mule3-agent#9]
Requires support in Java APM agent

**Comment by tomqwu** [comment]
@michaelhyatt , is this applying to the mule runtime manager which allowing multiple apps being deployed to the same server?

**Comment by michaelhyatt** [comment]
Yes, if you are using Mule domain-based projects, the APM agent configuration is done at the domain level and all the Mule domain projects are deployed and running as part of the same JVM will have the same service name, as per `elastic.apm.service_name` property.

So the workaround is to use more descriptive top-level flow names to create separate transactions for different Mule projects running within the same domain.

**Comment by tomqwu** [comment]
thank you, that's a relief for us so we know it will work in multiple domain apps under the same runtime agent container.

Should the APM agent lib dependencies and tracer XML spring bean be installed to domain level and only change the top-level flow name per app?

And lastly, do we need to put the config in the parent project instead of each app project?

**Comment by michaelhyatt** [comment]
Check out the sample app https://github.com/michaelhyatt/elastic-apm-mule3-agent/tree/master/example-domain-app

Everything is declared in the domain project, so dependant projects can remain tracer agnostic.
